modules/ml/src/ann_mlp.cpp

```diff
@@ -432,8 +432,12 @@ class ANN_MLPImpl : public ANN_MLP
                     double* data = sums.ptr<double>(i);
                     for( j = 0; j < cols; j++ )
                     {
-                        double t = scale2*(1. - data[j])/(1. + data[j]);
-                        data[j] = t;
+                        if(!cvIsInf(data[j])){
+                            double t = scale2*(1. - data[j])/(1. + data[j]);
+                            data[j] = t;
+                        }else{
+                            data[j] = -f_param2;
```

Comment:
I'd suggest to use `-scale2` here for consistency.
Please use same curly braces format as in the whole file:
```.cpp
if ()
{
    // ...
}
else
{
    // ...
}
```